stacks-signer/src/runloop.rs

```diff
@@ -175,8 +196,50 @@ impl<C: Coordinatable> RunLoop<C> {
         let inbound_messages: Vec<Packet> = event
             .modified_slots
             .iter()
+            // intercept and consume ping packets
+            .filter_map(|chunk| {
```

Comment:
I'm generally not a fan of deeply-nested code like this.  Can you refactor this so that (1) this behavior is encapsulated in its own method and (2) you keep the nesting / indentation level to a minimum?  Like, there's no reason to put the whole of the code body inside an `if is_ping_slot(..) { ... }` block.

Reply:
How do you find something in between?
```rust
            // intercept and consume ping packets
            .filter_map(|chunk| {
                let msg = LatencyPacket::verify_packet(chunk, self.signing_round.signer_id)?;

                match msg {
                    LatencyPacket::Pong(pong) => {
                        let id = pong.id();
                        // Signer won't react to Pongs from Pings not initiated by it.
                        self.ping_entries.get(&id).map(|tick| {
                            let variate = tick.elapsed();
                            info!("New RTT for id {id}: {:?}", variate);
                        });
                    }
                    LatencyPacket::Ping(ping) => {
                        let msg: LatencyPacket = ping.pong().into();
                        let _ = self
                            .stacks_client
                            .send_message_with_retry(signer_id, &msg)
                            .map(|ack| debug!("ACK: {:?}", ack))
                            .map_err(|e| warn!("Sending RTT probe failed! noop with error: {e}"));
                    }
                }
                None
            })
```
I can't find a zero-cost alternative to circumvent the fact that abstracting the last match into a method requires a `&mut self` such that stacks_client can increment the slot_version. The code above works because `self` is captured by the closure, and the Iterator is collected in the enclosing scope. As far as I know, it does not work with methods the same way as with closures. Taking that much effort to put the last match behind a method is not worth it. This logic can't be used anywhere else, and the match looks irreducible beyond moving it to its own method.
```rust
impl StacksClient {
    /// Sends messages to the stacker-db with an exponential backoff retry
    pub fn send_message_with_retry<'a, Packet>(
        &mut self,
        signer_id: u32,
        message: &'a Packet,
    ) -> Result<StackerDBChunkAckData, ClientError>
    where
        Packet: Serialize,
        &'a Packet: ToSlotId,
//snip ...
```

Reply:
> I can't find a zero-cost alternative to circumvent the fact that abstracting the last match into a method requires a &mut self such that stacks_client can increment the slot_version. The code above works because self is captured by the closure, and the Iterator is collected in the enclosing scope. As far as I know, it does not work with methods the same way as with closures. Taking that much effort to put the last match behind a method is not worth it. This logic can't be used anywhere else, and the match looks irreducible beyond moving it to its own method.

It is absolutely worth the time to make the code maintainable.  It is very straightforward to create a function that consumes the vector of messages, performs a "pass" on them to eliminate the pings, and returns the filtered vector.